### PR TITLE
[HALX86] Sort out MP HALs, Rearrange files

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Additional steps forward, per @vgalnt's comments.

JIRA issue: [CORE-17378](https://jira.reactos.org/browse/CORE-17378)

## Proposed changes

- [HALX86] CMakeLists.txt: Sort out MP HALs
Addendum to a064c5d.
- [HALX86] Rearrange files
Addendum to 7601fb5.
- [HALX86] generic/spinlock.c: Make #ifdef spans more explicit

### CODE REVIEW

File: hal/halx86/CMakeLists.txt [hunk 1]
```diff
@@ -57,17 +57,15 @@ if(ARCH STREQUAL "i386")
...
-    add_hal(halxbox SOURCES xbox/halxbox.rc COMPONENTS xbox up)
+    # add_hal(halmacpi SOURCES mp/halmp.rc COMPONENTS generic acpi mp apic)
+    # ReactOS port HALs:
     add_hal(halpc98 SOURCES pc98/halpc98.rc COMPONENTS pc98 up)
-
-    #add_hal(halmps SOURCES up/halup.rc COMPONENTS generic legacy smp pic)
-    #add_hal(halmacpi SOURCES acpi/halacpi.rc COMPONENTS generic acpi smp pic)
-    #add_hal(halmapic SOURCES acpi/halacpi.rc COMPONENTS generic legacy smp apic)
-    #add_hal(halmaacpi SOURCES acpi/halacpi.rc COMPONENTS generic acpi smp apic)
+    add_hal(halxbox SOURCES xbox/halxbox.rc COMPONENTS xbox up)
```
Comment (binarymaster) [inline hal/halx86/CMakeLists.txt:68]: Please move halpc98 and halxbox to `Legacy (non-ACPI) HALs`, as they are belong here.

**Review by binarymaster — COMMENTED** [review]
Also cc @disean for review, maybe you have some comments about this rearrangement.

File: hal/halx86/CMakeLists.txt (hunk 1 above)
Comment (SergeGautherie) [reply hal/halx86/CMakeLists.txt]: Moved.

**Comment by Extravert-ir** [comment]
Please, leave it to maintainers

**Pull request closed without merging**